File: roblox_plugin/src/Server/formatTypes/hash.ts
//!native
import type * as ProcessFormats from "../processFormats";
import SendMessagesThread from "../sendMessages";
import ReceiveMessagesThread from "../receiveMessages";

const DATA_SIZE = 32;
export function read(
	receiveThread: ReceiveMessagesThread,
	args: defined[],
	formatData: ProcessFormats.FormatData,
	masks: Map<string, boolean>,
) {
	const [readBuffer, readOffset] = ReceiveMessagesThread.readBuffer(receiveThread, DATA_SIZE);
	const hashValue = buffer.readstring(readBuffer, readOffset, DATA_SIZE);
	args.push(hashValue);
}
export function write(
	sendThread: SendMessagesThread,
	args: defined[],
	argsCount: number,
	formatData: ProcessFormats.FormatData,
	masks: Map<string, boolean>,
): number {
	const hashValue = args[argsCount] as string;
	const data = buffer.fromstring(hashValue);
	SendMessagesThread.writeBuffer(sendThread, data, DATA_SIZE);
	return argsCount + 1;
}

File: blender_roblox_sync/roblox_interface/server/format_types/hash.py
format_size = 32
def read(self, format_data, **masks):
	read_buffer, read_offset = self.read_buffer(format_size)
	return (read_buffer[read_offset:read_offset + format_size],)

def write(self, args, args_count, format_data, **masks):
	hash_value = args[args_count]
	self.write_buffer(hash_value, format_size)
	return args_count + 1